### PR TITLE
1559: Added LLM services module

### PR DESCRIPTION
**Pull request by cableman**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/1559

#### Description

* Front end for chat with LLM services.

#### Screenshot of the result

N/A

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist, you have to add a comment explaining why this change
should be exempt from the list.

#### Additional comments or questions

N/A

### CODE REVIEW

**Comment by cableman** [comment]
To reviewer: The template part is going to change. So don't look at that part.

File: web/modules/custom/chat/src/Plugin/Block/ChatBlock.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+      '#min' => 0,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockValidate($form, FormStateInterface $form_state): void {
```
Comment (jekuaitk) [inline web/modules/custom/chat/src/Plugin/Block/ChatBlock.php:191]: We could potentially add a check for `top_k` being a positive integer.

File: web/modules/custom/chat/js/chat.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+(function ($, Drupal, once) {
+  Drupal.behaviors.my_custom_behavior = {
```
Comment (jekuaitk) [inline web/modules/custom/chat/js/chat.js:2]: This should probably be renamed

File: web/modules/custom/chat/src/Plugin/Block/ChatBlock.php (hunk 1 above)
Comment (cableman) [reply web/modules/custom/chat/src/Plugin/Block/ChatBlock.php:191]: Ahh, well spotted... the `#min` is missing for that.

**Review by jekuaitk — APPROVED** [review]

**Comment by cableman** [comment]
Rebased with develop

**Pull request merged**